### PR TITLE
compaction: add ability to set max compaction level

### PR DESCRIPTION
**Pull request by xjewer**
On an overloaded scrapers the amount of samples is enormous. Block size
on the 3d compaction level could be more than 30G, hence the next compaction
takes a big effort to download, compact and upload of hundreds Gb (hundreds of billions samples).
This could be essential at least for the undergoing development.

## Verification

```
--debug.max-compaction-level=4  Maximum compaction level of presets, max is 4: 0=1h, 1=2h, 2=8h, 3=48h, 4=336h
```

### CODE REVIEW

**Comment by bwplotka** [comment]
What do you mean by `This could be essential at least for the undergoing development.` ? (:

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -34,6 +68,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	objStoreConfig := regCommonObjStoreFlags(cmd, "")
 
+	currentMaxLevel := len(compactions) - 1
+	maxCompactionLevel := cmd.Flag("compaction.max-level", fmt.Sprintf("Maximum compaction level, default is %d: %s", currentMaxLevel, compactions.String())).Default(strconv.Itoa(currentMaxLevel)).Int()
```
Comment (bwplotka) [inline cmd/thanos/compact.go]: Let's make it Hidden:
```suggestion
	maxCompactionLevel := cmd.Flag("compaction.max-level", fmt.Sprintf("Maximum compaction level, default is %d: %s", currentMaxLevel, compactions.String())).Default(strconv.Itoa(currentMaxLevel)).Hidden().Int()
```
as this is not recommended to change this.

File: cmd/thanos/compact.go [hunk 2]
```diff
@@ -21,6 +23,38 @@ import (
...
+
+func (cs compactionSet) String() string {
+	result := make([]string, len(cs))
+	for i, c := range cs {
+		result[i] = fmt.Sprintf("%d=%dh", i, int(c.Hours()))
+	}
+	return strings.Join(result, ", ")
+}
+
+func (cs compactionSet) enabled(maxLevel int) ([]int64, error) {
```
Comment (bwplotka) [inline cmd/thanos/compact.go]: Why `enabled` verb? what we enable here? Maybe `levels` instead?

```suggestion
func (cs compactionSet) levels(maxLevel int) ([]int64, error) {
```

File: cmd/thanos/compact.go (hunk 1 above)
Comment (xjewer) [reply cmd/thanos/compact.go]: 👌

File: cmd/thanos/compact.go (hunk 2 above)
Comment (xjewer) [reply cmd/thanos/compact.go]: assumed enabled levels, but levels instead should be ok

**Pull request merged**